extensions/amp-ad/0.1/amp-ad-3p-impl.js

```diff
@@ -305,10 +311,18 @@ export class AmpAd3PImpl extends AMP.BaseElement {
     user().assert(!this.isInFixedContainer_,
         '<amp-ad> is not allowed to be placed in elements with ' +
         'position:fixed: %s', this.element);
-    this.layoutPromise_ = getAdCid(this).then(cid => {
+
+    const consentPolicyId = super.getConsentPolicy();
+    const consentPromise = consentPolicyId
+      ? getConsentPolicyState(this.getAmpDoc(), consentPolicyId)
+      : Promise.resolve(null);
+
+    this.layoutPromise_ = Promise.all(
```

Comment:
`this.layoutPromise_` is blocked by the consent. However creating iframe and `this.xOriginIframeHandler_.init(iframe);` is not. This breaks the consent blocking behavior here. Let's fix this and patch to prod...